functions/Get-DbaDbRole.ps1

```diff
@@ -0,0 +1,137 @@
+	<#
+.SYNOPSIS
+Get database roles on a Sql instance.
+
+.DESCRIPTION
+Get database roles on a Sql instance.
+
+Default output includes columns SQLServer, Database, Role.
+
+.PARAMETER SQLInstance
+The SQL Server that you're connecting to.
+
+.PARAMETER SqlCredential
+Allows you to login to servers using SQL Logins as opposed to Windows Auth/Integrated/Trusted. To use:
+
+$scred = Get-Credential, then pass $scred object to the -SqlCredential parameter. 
+
+SQL Server does not accept Windows credentials being passed as credentials. To connect as a different Windows user, run PowerShell as that user.
+
+.PARAMETER Database
+The database(s) to process - this list is auto-populated from the server. If unspecified, all databases will be processed.
+
+.PARAMETER ExcludeDatabase
+The database(s) to exclude - this list is auto-populated from the server
+
+.PARAMETER NoFixedRole
+Excludes all fixed roles.
+
+.PARAMETER Credential
+Credential object used to connect to the SQL Server as a different user.
+
+.PARAMETER EnableException
+	By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+	This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+	Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+.NOTES
+Tags: Roles, Databases
+Author: Klaas Vandenberghe ( @PowerDBAKlaas )
+
+Website: https://dbatools.io
+Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+.LINK
+ https://dbatools.io/Get-DbaDbRole
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerA
+
+Returns a custom object displaying SQLServer, Database, Role for all DatabaseRoles on sql instance ServerA.
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerA | Out-Gridview
+
+Returns a gridview displaying SQLServer, Database, Role for all DatabaseRoles on sql instance ServerA.
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerB\sql16 -ExcludeDatabase DBADB,TestDB
+
+Returns a gridview displaying SQLServer, Database, Role for DatabaseRoles on sql instance ServerB\sql16, except those in databases DBADB and TestDB.
+
+.EXAMPLE
+'ServerB\sql16','ServerA' | Get-DbaDbRole
+
+Returns a gridview displaying SQLServer, Database, Role for DatabaseRoles on sql instances ServerA and ServerB\sql16.
+
+.EXAMPLE
+Get-DbaDbRole -SqlInstance ServerB\sql16 -Database AccountingDB
+
+Returns a gridview displaying SQLServer, Database, Role for DatabaseRoles in database AccountingDB on sql instance ServerB\sql16.
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Mandatory, ValueFromPipeline)]
+		[Alias('SqlServer', 'ServerInstance')]
+		[DbaInstanceParameter[]]$SqlInstance,
+		[Alias("Credential")]
+		[PSCredential]$SqlCredential,
+		[object[]]$Database,
+		[object[]]$ExcludeDatabase,
+		[switch]$NoFixedRole,
+		[switch][Alias('Silent')]$EnableException
+	)
```

Comment:
Do we want to add the ability to pass in a "user" and/or "login"? This would provide support for piping from maybe `Get-DbaDbUser` or `Get-DbaLogin`.

Reply:
I think that's rather suited for a `Test-DbaRoleMemberShip` or something.
If we include such functionality in here, we could pipe user or table or owner to `Get-DbaDatabase` and many more.
I would prefer those 'base' functions just return the objects.